### DISCUSSION TITLE
Non-fault policies/strategies (e.g. polling)

### DISCUSSION THREAD

**Discussion by kmcclellan** [App-vNext/Polly#1373] [Ideas]
Any thought given to expanding this library to express execution policies/strategies that don't involve exceptions? A simple example is a poller that runs periodically using a random decorrelation coefficient. I've found myself wanting to leverage the configurable "backoff" delay + random jitter implemented in this library in scenarios outside the realm of fault handling.

**Comment by martincostello** [comment]
This is supported via methods such as `HandleResult<T>` as [documented in the README](https://github.com/App-vNext/Polly#step-1b-optionally-specify-return-results-you-want-to-handle).

For example, you could retry until the result of a method was `true`.
